src/psyclone/generator.py

```diff
@@ -428,22 +428,44 @@ def main(args):
         psy_str = str(psy)
         alg_str = str(alg)
     if args.oalg is not None:
-        my_file = open(args.oalg, "w")
-        my_file.write(alg_str)
-        my_file.close()
+        write_unicode_file(alg_str, args.oalg)
     else:
         print("Transformed algorithm code:\n%s" % alg_str)
 
     if not psy_str:
         # empty file so do not output anything
         pass
     elif args.opsy is not None:
-        my_file = open(args.opsy, "w")
-        my_file.write(psy_str)
-        my_file.close()
+        write_unicode_file(psy_str, args.opsy)
     else:
         print("Generated psy layer code:\n", psy_str)
 
 
+def write_unicode_file(contents, filename):
+    '''
+    Wrapper routine that ensures that a string is encoded as unicode before
+    writing to file in both Python 2 and 3.
+
+    :param str contents: string to write to file.
+    :param str filename: the name of the file to create.
+
+    '''
+    import six
+    import io
+
+    if six.PY2:
+        # In Python 2 the plain string must be encoded as unicode for the call
+        # to write() below. unicode() does not exist in Python 3 since all
+        # strings are unicode.
+        # pylint: disable=undefined-variable
```

Comment:
I can't see the purpose for the pylint disable here. My pylint does not complain.

Reply:
It's because there's no `unicode` in Python 2.

Reply:
OK, I believe you, but I can't get pylint to complain in either Python2 or 3 with these pylint directives removed.